### PR TITLE
Add basic infrastructure for building, testing, and code quality checks

### PR DESCRIPTION
**Pull request by smarr**
This PR introduces:

- a minimal test example of Truffle code to check the infrastructure
- a script to maintain Graal/Truffle/MX dependencies avoiding a submodule (using a submodule would be problematic for languages that use submodules)
- the build system is based on Ant as for the other SOM projects (didn't think that it is necessary to change this)
- added an Eclipse project to have common settings everywhere
- added CI settings for Travis as well as GitLab CI

### CODE REVIEW

File: tests/bd/testsetup/TestAdd.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.junit.Test;
+
+import com.oracle.truffle.api.nodes.UnexpectedResultException;
+
+
+public class TestAdd {
+
+  @Test
+  public void test() throws UnexpectedResultException {
```
Comment (charig) [inline tests/bd/testsetup/TestAdd.java]: I would just name it testLiteralIntegerAddition

**Review by charig — APPROVED** [review]

File: tests/bd/testsetup/TestAdd.java (hunk 1 above)
Comment (smarr) [reply tests/bd/testsetup/TestAdd.java]: Fixed, thanks.

File: build.xml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        <available file="graal/.git" property="graal.present"/>
+    </target>
+    <target name="graal-repo" depends="check-graal-available"
+            unless="graal.present">
+      <exec executable="${basedir}/.graal-git-repo">
+        <arg value="checkout" />
+      </exec>
+    </target>
+
+    <target name="graal" depends="graal-repo">
```
Comment (charig) [inline build.xml:90]: Why named graal? Isn't this compiling truffle?

Comment (smarr) [reply build.xml:90]: At the moment, it is only compiling Truffle, yes.
But we might need more, for instance the Graal SDK stuff, don't know yet.
So, I just named the target after the repo, which is called 'graal'.

**Pull request merged**